domain/src/main/java/com/github/maleksandrowicz93/educational/institution/PersonName.java

```diff
@@ -0,0 +1,12 @@
+package com.github.maleksandrowicz93.educational.institution;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+
+@Builder(access = AccessLevel.PACKAGE)
+record PersonName(
+        Name firstName,
+        Name secondName,
```

Comment:
nie każdy ma drugie imię, a API tego rekordu tego nie obrazuje

Reply:
Builder obrazuje - można zbudować obiekt z dowolnej ilości pól. Wydaje mi się, że zastosowanie kilku konstruktorów utworzy zbędny kod, bo i tak ich ekwiwalent mamy w builderze.

Reply:
builder domyślnie nie utrzymuje reguł poprawności, a value object powinien weryfikować poprawność danych i w swoim API prezentować to, czym taka poprawność jest

żeby miał tutaj sens builder, powinien wyglądać np. tak:

```java
ParsonName.create(firstName, lastName)
          .withMiddleName(middleName)
          .build();
```

co nie ma żadnej przewagi, a nawet traci IMO względem

```java
new ParsonName(firstName, lastName);
new ParsonName(firstName, middleName, lastName);
```

API odzwierciedla to, co dany obiekt reprezentuje